device/device.go

```diff
@@ -72,50 +74,152 @@ type Device struct {
 	OnConnectionStateChanged    func(*Device, bool)
 }
 
-// NewDevice creates a new Device
-func NewDevice(deviceID, realm, credentialsSecret string, pairingBaseURL string) (*Device, error) {
-	// Create temporary directory for the persistent data
-	// TODO: How to clean this up?
-	persistencyDir, err := ioutil.TempDir("", deviceID)
-	if err != nil {
-		return nil, err
+// DeviceOptions define a set of advanced options when creating the Device
+type DeviceOptions struct {
```

Comment:
Some arguably advanced options (e.g. `MaxInflightMessages`) are left out of this struct. I think we should either expand the struct to include them, or rename `DeviceOptions` to something more specific, like `PersistencyOptions`

Reply:
We should definitely include them here IMHO, this should become a single endpoint for all configuration unless we want to take a different approach.